lib/Backends/OpenCL/Transforms.cpp

```diff
@@ -50,17 +50,92 @@ bool OCLBackend::transformPostLowering(Function *F,
       continue;
     }
     if (auto *PMN = dyn_cast<MaxPoolNode>(&node)) {
+      if (cctx.compMode == CompilationMode::Train)
+        continue;
+
       auto *NR = convertPoolToNCHWPool<MaxPoolNode, OCLMaxPoolNode>(PMN, F);
       PMN->getResult().replaceAllUsesOfWith(NR);
       changed = true;
       continue;
     }
     if (auto *PAN = dyn_cast<AvgPoolNode>(&node)) {
+      if (cctx.compMode == CompilationMode::Train)
+        continue;
+
       auto *NR = convertPoolToNCHWPool<AvgPoolNode, OCLAvgPoolNode>(PAN, F);
       PAN->getResult().replaceAllUsesOfWith(NR);
       changed = true;
       continue;
     }
+    if (auto *BRA = dyn_cast<BatchedReduceAddNode>(&node)) {
```

Comment:
Could you add a comment before this `if` describing what it does, i.e. it replaces Glow's BatchedReduceAddNode by an OpenCL specific OCLBatchedReduceAddNode, because ...